### PR TITLE
Implement countdown mode for DatePickerIOS

### PR DESCRIPTION
**Pull request by dapetcu21**
I needed this for my project and I thought I'd share the implementation. I'm open to shotgun fire :smile:

### CODE REVIEW

File: Libraries/Components/DatePicker/DatePickerIOS.ios.js [hunk 1]
```diff
@@ -97,6 +112,25 @@ var DatePickerIOS = React.createClass({
     };
   },
 
+  _onChangeCountDownDuration: function(event: Event) {
+    var nativeDate = new Date(event.nativeEvent.timestamp);
+    var nativeDuration = nativeDate.getMinutes() * 60 + nativeDate.getHours() * (60 * 60);
```
Comment (dapetcu21) [inline Libraries/Components/DatePicker/DatePickerIOS.ios.js]: Is there a better way of getting this? Like, maybe getting it from the Objective-C view?

**Comment by mkonicek** [comment]
Hey @dapetcu21, would you be up for releasing the countdown picker as an npm module to get feedback?
See https://js.coach

Not sure how many people need this feature in React Native core.

**Comment by dapetcu21** [comment]
Hi. Yeah. I could do that. Dunno exactly when, but I hope I'll catch some free time this week.

The reason why I think this should be in core is because this is just exposing the functionality of an underlying Cocoa component that already is in the core.

**Comment by facebook-github-bot-4** [comment]
@sahrens would you mind taking a look at this pull request? It's been a while since the last commit was reviewed.

**Comment by chirag04** [comment]
@dapetcu21 @mkonicek I made a module for countdown picker here: https://github.com/4Catalyzer/react-native-countdown-picker

**Comment by mkonicek** [comment]
OK, let's close this one.

**Comment by mkonicek** [comment]
c

**Comment by mkonicek** [comment]
c

**Comment by mkonicek** [comment]
c

**Comment by mkonicek** [comment]
c

**Comment by mkonicek** [comment]
c

**Comment by mkonicek** [comment]
c

**Comment by jarrx** [comment]
This solution seems to work quite well but I'm experiencing a bug where there's no event firing the first time you set a value. The second time, it works fine. This seems to be an old iOS bug, see [https://github.com/romaonthego/RETableViewManager/issues/161](https://github.com/romaonthego/RETableViewManager/issues/161), but I'm seeing this even though I'm running on iOS 11.2. Anyone with native experience who has a solution to this?

**Pull request closed without merging**